modular_skyrat/modules/vore/code/__HELPERS/vore_helpers.dm

```diff
@@ -0,0 +1,106 @@
+/proc/vore_replace(messages, mob/living/pred=null, mob/living/prey=null, belly=null)
+	if (!istext(messages) && !(islist(messages) && length(messages)))
+		return ""
+	var/message = istext(messages) ? messages : pick(messages)
+	var/list/replacements = list("%pred" = "[pred]", "%prey" = "[prey]", "%belly" = "[belly]")
+	for (var/replacement in replacements)
+		message = replacetext(message, replacement, replacements[replacement])
+	return message
+
+//syntax: %a|what the text seen by the audience is|what the text seen by the pred is|what the text seen by the prey is|
+//ie "%a|pred_here|You|pred_here| %a|manages|manage|manages| to eat %a|prey_here|prey_here|you|!"
+/proc/person_vore_replace(message, mob/target, mob/living/pred=null, mob/living/prey=null)
+	var/static/regex/audience_replace = regex(@"%a\|([\s\S]*?)\|[\s\S]*?\|[\s\S]*?\|", "g")
+	var/static/regex/pred_replace = regex(@"%a\|[\s\S]*?\|([\s\S]*?)\|[\s\S]*?\|", "g")
+	var/static/regex/prey_replace = regex(@"%a\|[\s\S]*?\|[\s\S]*?\|([\s\S]*?)\|", "g")
```

Comment:
This is very coomplicated regex and with very dubious usefulness. I'm not sure why this would be better than manually typing out the messages, as this seemingly never gets used by the player.

Reply:
mostly because I couldn't think of a good way to send a message to the audience, to the pred, and to the prey while respecting prefs, so I just made my own function and had all the messages built in. It never gets used by the player but it does get used by the code, and it looks nicer imo